src/index.js

```diff
@@ -2,31 +2,39 @@ var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
 var API = 'https://rickandmortyapi.com/api/character/';
 var xhttp = new XMLHttpRequest();
+var dataBox = [];
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+function fetchData(url_api, data){
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = (event) =>{
+      if (xhttp.readyState === 4 && xhttp.status == 200){
+        resolve(JSON.parse(xhttp.responseText), data);
+      } else return reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```